web/src/message_lists.ts

```diff
@@ -0,0 +1,85 @@
+import $ from "jquery";
+
+import * as blueslip from "./blueslip";
+import * as inbox_util from "./inbox_util";
+import * as recent_view_util from "./recent_view_util";
+import * as ui_util from "./ui_util";
+import type {Message} from "./types";
+
+// TODO(typescript): Move this to message_list_view when it's
+// converted to typescript.
+type MessageListView = {
+    update_recipient_bar_background_color: () => void;
+    _render_win_start: any;
+    _render_win_end: any;
+};
+
+type MessageList = {
+    table_name: string;
+    view: MessageListView;
+    selected_id: () => number | undefined;
+    selected_row: () => JQuery | undefined;
+    selected_idx: () => number;
+    all_messages: () => Message[];
+    get: (selected_id: number| undefined) => Message;
+    pre_narrow_offset?: number;
+};
+
+export let home: MessageList | undefined;
+export let current : MessageList | undefined;
+
+export function set_current(msg_list: MessageList): void {
+    current = msg_list;
+}
+
+export function set_home(msg_list: MessageList): void {
+    home = msg_list;
+}
+
+export function all_rendered_message_lists(): MessageList[] {
+    const rendered_message_lists = [];
+    if (home) {
```

Comment:
There's a lot of code I had to tweak because `home` and `current` can sometimes be `undefined` (presumably only before the initialization)

Reply:
In general we want to throw errors on invariant violations, rather than silently ignoring them.

Reply:
Okay cool, switched to that. Is that the kind of errors we want?

Reply:
Oh I was just thinking `assert(home !== undefined)`; no need to waste our oratory skills on messages.

Reply:
Ohhh yeah of course. Thanks